ycmd/completers/language_server/language_server_completer.py

```diff
@@ -700,6 +700,42 @@ def _GoToDeclaration( self, request_data ):
       )
 
 
+  def _GoToReferences( self, request_data ):
+    request_id = self.GetServer().NextRequestId()
+    response = self.GetServer().GetResponse( request_id,
+                                             lsapi.References( request_id,
+                                                               request_data ) )
+
+    if isinstance( response[ 'result' ], list ):
+      if len( response[ 'result' ] ) > 1:
+        positions = response[ 'result' ]
+        return [
+          responses.BuildGoToResponseFromLocation(
+            # TODO: Codepoint to byte offset
+            responses.Location(
+              position[ 'range' ][ 'start' ][ 'line' ] + 1,
+              position[ 'range' ][ 'start' ][ 'character' ] + 1,
+              lsapi.UriToFilePath( position[ 'uri' ] ) )
+          ) for position in positions
+        ]
+      else:
+        position = response[ 'result' ][ 0 ]
+        return responses.BuildGoToResponseFromLocation(
+          # TODO: Codepoint to byte offset
+          responses.Location( position[ 'range' ][ 'start' ][ 'line' ] + 1,
+                              position[ 'range' ][ 'start' ][ 'character' ] + 1,
+                              lsapi.UriToFilePath( position[ 'uri' ] ) )
+        )
+    else:
```

Comment:
According to [the protocol](https://github.com/Microsoft/language-server-protocol/blob/master/protocol.md#find-references-request), the references response must be a list, so I don't think we need this else clause.

Reply:
Should I leave the `if isinstance( response[ 'result' ], list)` part? If so and I remove the whole `else` block then, in case something goes wrong and the result is not a list we will get an error. In other words, should I leave something like:
```
else:
  return []
```

Reply:
As above, a non-list reply would be invalid (according to protocol)

Reply:
Done.